community/index/src/main/java/org/neo4j/index/gbptree/TreeNode.java

```diff
@@ -370,6 +371,26 @@ int readChildrenWithInsertRecordInPosition( PageCursor cursor, Consumer<PageCurs
                 childSize(), childOffset( 0 ), into );
     }
 
+    void goTo( PageCursor cursor, long childId, long stableGeneration, long unstableGeneration )
+            throws IOException
+    {
+        if ( !cursor.next( childId ) )
+        {
+            throw new IllegalStateException( "Could not go to " + childId );
```

Comment:
This error message could be improved. If this pops up in front a user, they'd have no idea what to do. The `childId` should at least have a unit associated with it, and we could qualify the message with a `GBPTree Index Structure Error: ` prefix. Perhaps you should make a custom runtime exception subclass for these runtime index exceptions. It would also be nice to attach which specific index the error relates to, in case people want to try to remedy the situation with a drop+create index command sequence.